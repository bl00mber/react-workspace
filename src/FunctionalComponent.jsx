import React from 'react';
import PropTypes from 'prop-types';

const FunctionalComponent = props => (
  <h3>{props.name}</h3>
);

FunctionalComponent.propTypes = {
  name: PropTypes.string.isRequired,
};

export default FunctionalComponent;
