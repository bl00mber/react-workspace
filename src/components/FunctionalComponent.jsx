import React from 'react';
import PropTypes from 'prop-types';

const FunctionalComponent = ({ name }) => (
  <h3>{name}</h3>
);

FunctionalComponent.propTypes = {
  name: PropTypes.string.isRequired,
};

export default FunctionalComponent;
