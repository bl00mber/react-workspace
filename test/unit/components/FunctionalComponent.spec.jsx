import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import FunctionalComponent from '@/components/FunctionalComponent';

describe('FunctionalComponent', () => {
  test('should match enzyme shallow snapshot', () => {
    const wrapper = shallow(<FunctionalComponent name="Class Component" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
