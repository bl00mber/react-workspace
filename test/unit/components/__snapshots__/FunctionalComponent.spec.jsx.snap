// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FunctionalComponent should match enzyme shallow snapshot 1`] = `
<h3>
  Class Component
</h3>
`;
